Content.Client/GameObjects/Components/Mobs/ClientStatusEffectsComponent.cs

```diff
@@ -80,32 +85,74 @@ private void PlayerDetached()
             _ui = null;
         }
 
-        public void UpdateIcons()
+        public void UpdateStatusEffects()
         {
             if (!CurrentlyControlled || _ui == null)
             {
                 return;
             }
+            _cooldown.Clear();
             _ui.VBox.DisposeAllChildren();
 
-            foreach (var effect in _icons.OrderBy(x => (int) x.Key))
+            foreach (var effect in _status.OrderBy(x => (int) x.Key))
             {
-                TextureRect newIcon = new TextureRect
+                var cooldown = new CooldownGraphic();
```

Comment:
So this creates a new cooldown for even status effects with no cooldown values, right? Would that be avoidable?

Reply:
Yeah, it could be avoided. Now it's only created if needed.